### PR TITLE
Update signature verification docs for Sigstore 2.0

### PR DESCRIPTION
**Pull request by jedsalazar**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

Fixes: updates to cosign 2.0 for Cilium signature verification

```release-note
<!-- Enter the release note text here if needed or remove this section! -->
```

### CODE REVIEW

File: Documentation/configuration/verify-image-signatures.rst [hunk 1]
```diff
@@ -26,14 +26,13 @@ Let's verify a Cilium image's signature using the ``cosign verify`` command:
 
 .. code-block:: shell-session
 
-    $ COSIGN_EXPERIMENTAL=1 cosign verify --certificate-github-workflow-repository cilium/cilium --certificate-oidc-issuer https://token.actions.githubusercontent.com --certificate-github-workflow-name "Image Release Build" --certificate-github-workflow-ref refs/tags/[RELEASE TAG] quay.io/cilium/cilium:v1.13 | jq
+    $ cosign verify --certificate-github-workflow-repository cilium/cilium --certificate-oidc-issuer https://token.actions.githubusercontent.com --certificate-github-workflow-name "Image Release Build" --certificate-github-workflow-ref refs/tags/[RELEASE TAG] quay.io/cilium/cilium:v1.13 --certificate-identity "https://github.com/cilium/cilium/.github/workflows/build-images-releases.yaml@refs/tags/v1.13.0" | jq
```
Comment (aanm) [inline Documentation/configuration/verify-image-signatures.rst]:
```
    $ TAG=v1.13.0
    $ cosign verify --certificate-github-workflow-repository cilium/cilium \
    --certificate-oidc-issuer https://token.actions.githubusercontent.com \
    --certificate-github-workflow-name "Image Release Build" \
    --certificate-github-workflow-ref refs/tags/${TAG} \
    --certificate-identity "https://github.com/cilium/cilium/.github/workflows/build-images-releases.yaml@refs/tags/${TAG}" \
    "quay.io/cilium/cilium:${TAG}" | jq
```

**Review by aanm — CHANGES_REQUESTED** [review]
Thank you @jedsalazar, can we re-format the command so it's more readable in our docs?

File: Documentation/configuration/verify-image-signatures.rst [hunk 2]
```diff
@@ -26,14 +26,13 @@ Let's verify a Cilium image's signature using the ``cosign verify`` command:
...
     
 
 .. note::
 
-    ``COSIGN_EXPERIMENTAL=1`` is used to allow verification of images signed in 
-    ``KEYLESS`` mode. To learn more about keyless signing, please refer to 
-    `Keyless Signatures`_.
+    ``cosign`` is using verifying images signed in ``KEYLESS`` mode. To learn
```
Comment (qmonnet) [inline Documentation/configuration/verify-image-signatures.rst]: ` ``cosign`` is using verifying images signed in ``KEYLESS`` mode.`

-> Should this be `... is used to verify images...` instead?

(Hi Jed! :wave:)

File: Documentation/configuration/verify-image-signatures.rst (hunk 1 above)
Comment (jedsalazar) [reply Documentation/configuration/verify-image-signatures.rst]: Thanks, Andre. Looks much better!

File: Documentation/configuration/verify-image-signatures.rst (hunk 2 above)
Comment (jedsalazar) [reply Documentation/configuration/verify-image-signatures.rst]: Hi, Quentin! Thanks for the feedback =)

**Comment by jedsalazar** [comment]
Updated changes to address feedback. Thanks for having a look @aanm and @qmonnet!

**Review by aanm — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]
Thanks!

**Review by michi-covalent — APPROVED** [review]

**Pull request merged**